torchreid/integration/nncf/compression.py

```diff
@@ -0,0 +1,233 @@
+        raise RuntimeError(f'Only ONE of datamanager_for_init or checkpoint_path should be set: '
+                           f'datamanager_for_init={datamanager_for_init} checkpoint_path={checkpoint_path}')
+
+    h, w = cfg.data.height, cfg.data.width
+    if not nncf_config_data:
+        print('Using the default NNCF int8 quantization config')
+        nncf_config_data = get_default_nncf_compression_config(h, w)
+
+    # do it even if nncf_config_data is loaded from a checkpoint -- for the rare case when
+    # the width and height of the model's input was changed in the config
+    # and then finetuning of NNCF model is run
+    nncf_config_data.setdefault('input_info', {})
+    nncf_config_data['input_info']['sample_size'] = [1, 3, h, w]
+
+    nncf_config = NNCFConfig(nncf_config_data)
+    print(f'nncf_config =\n{pformat(nncf_config)}')
+    if not nncf_metainfo:
+        nncf_metainfo = create_nncf_metainfo(nncf_compression_enabled=True,
+                                             nncf_config=nncf_config_data)
+    else:
+        # update it just to be on the safe side
+        nncf_metainfo['nncf_config'] = nncf_config_data
+
+    class ReidInitializeDataLoader(InitializingDataLoader): #TODO: check is it correct
+        def get_inputs(self, dataloader_output):
+            # define own InitializingDataLoader class using approach like
+            # parse_data_for_train and parse_data_for_eval in the class Engine
+            # dataloader_output[0] should be image here
+            args = (dataloader_output[0], )
+            return args, {}
+
+    cur_device = next(model.parameters()).device
+    print(f'NNCF: cur_device = {cur_device}')
+
+    if checkpoint_path is None:
+        print('No NNCF checkpoint is provided -- register initialize data loader')
+        train_loader = datamanager_for_init.train_loader
+        wrapped_loader = ReidInitializeDataLoader(train_loader)
+        nncf_config = register_default_init_args(nncf_config, wrapped_loader, device=cur_device)
+        resuming_state_dict = None
+    else:
+        print(f'Loading NNCF model from {checkpoint_path}')
+        resuming_state_dict = _load_checkpoint_for_nncf(model, checkpoint_path, map_location=cur_device)
+        print(f'Loaded NNCF model from {checkpoint_path}')
+
+    transform = build_inference_transform(
+        cfg.data.height,
+        cfg.data.width,
+        norm_mean=cfg.data.norm_mean,
+        norm_std=cfg.data.norm_std,
+    )
+
+    def dummy_forward(model):
+        prev_training_state = model.training
+        model.eval()
+        input_img = random_image(cfg.data.height, cfg.data.width)
+        input_blob = transform(input_img).unsqueeze(0)
+        assert len(input_blob.size()) == 4
+        input_blob = input_blob.to(device=cur_device)
+        input_blob = nncf_model_input(input_blob)
+        model(input_blob)
+        model.train(prev_training_state)
+
+    # TODO: think if this is required
```

Comment:
Have you already solved your TODO? Or it's not important?

Reply:
Removed one of TODO, reworded the other one.

Now I won't change this part of code any more.